frappe/app.py

```diff
@@ -318,7 +321,10 @@ def handle_exception(e):
 	return response
 
 
-def after_request(rollback):
+def after_request(rollback: bool) -> bool:
+	for after_request_task in frappe.get_hooks("after_request"):
+		frappe.call(after_request_task)
```

Comment:
`after_request` wont be called in case of exceptions. Maybe this should be in `finally` block?